src/wasm.rs

```diff
@@ -134,3 +138,78 @@ impl Server {
     pub fn register_tag_keys_callback(&mut self, _f: Function) {}
     pub fn register_tag_values_callback(&mut self, _f: Function) {}
 }
+
+// Flux crate packages migrated over
+/// (Generated by WASM.)
+#[wasm_bindgen]
+pub fn parse(s: &str) -> JsValue {
+    let mut p = Parser::new(s);
+    let file = p.parse_file(String::from(""));
+
+    JsValue::from_serde(&file).unwrap()
+}
+
+/// Format a JS file.
+#[wasm_bindgen]
+pub fn format_from_js_file(js_file: JsValue) -> String {
+    if let Ok(file) = js_file.into_serde::<File>() {
+        if let Ok(converted) = convert_to_string(&file) {
+            return converted;
+        }
+    }
+    "".to_string()
```

Comment:
I realize you're just copy/pasting here, but if there's an error, does this just return an empty string (possibly deleting the contents of the file in the first place)? Do I read that right?

Reply:
That is correct. I've had conversations with folks on the UI team about this function before, and it has been a pain point for them.

Reply:
Earlier I changed it to a format that outputted the error per Sean's recommendation for debugging purposes: 
#[wasm_bindgen]
pub fn format_from_js_file(js_file: JsValue) -> String {
    let s = match js_file.into_serde::<File>() {
        Ok(file) => match convert_to_string(&file) {
            Ok(converted) => converted,
            Err(e) => e.to_string(),
        },
        Err(e) => e.to_string(),
    };
    s
}
Should I just change it to something like this?

Reply:
I already mentioned this to @BarWeiner1 in a private chat, but returning an error as a plain string is not best practice. Unfortunately, the need to ship wasm code places limitations on the kind of error handling we can do. Not sure what the fix is here.

Reply:
I think the fix should be to return whatever flux was sent to it. I haven't looked too hard at what that would entail, but the indicator is more or less "I can't format this code because I don't think it's flux code" is probably the correct thing.